src/modules/infobox/components/FeatureProfilePlot.vue

```diff
@@ -0,0 +1,338 @@
+            const tooltipCenterX = Math.min(
+                Math.max(tooltipHalfWidth - profilePlotMargin.left, x),
+                plotWidth - tooltipHalfWidth + profilePlotMargin.right
+            )
+            // done like this because using of computed makes it very slow X position of arrow (relative to tooltip)
+            this.$refs.profileTooltipArrow.style.left =
+                tooltipHalfWidth + (x - tooltipCenterX) + 'px' // X position of the tooltip center
+            profileTooltipElement.style.left =
+                tooltipCenterX +
+                profilePlotMargin.left +
+                profilePlot.getBoundingClientRect().x +
+                'px'
+            // Y position of arrowhead
+            profileTooltipElement.style.top =
+                y + profilePlotMargin.top + profilePlot.getBoundingClientRect().y + 'px'
+            // Tooltip text
+            profileTooltipElement.querySelector(
+                '.distance'
+            ).innerText = `${distanceUnderMouseCursor.toFixed(2)} m`
+            profileTooltipElement.querySelector(
+                '.elevation'
+            ).innerText = `${elevationUnderMouseCursor.toFixed(2)} m`
+
+            // Update the position of the overlay
+            const coords = proj4(
+                CoordinateSystems.LV95.epsg,
+                CoordinateSystems.WEBMERCATOR.epsg,
+                this.profileData.lineString.getCoordinateAt(x / plotWidth)
+            )
+            this.currentHoverPosOverlay.setPosition(coords)
+        },
+        onMouseOutOverProfile() {
+            this.showTooltip = false
+            this.getMap().removeOverlay(this.currentHoverPosOverlay)
+        },
+
+        /**
+         * Given a distance from origin, calculate the height.
+         *
+         * @param {Number} dist Distance (must be in the unit specified by
+         *   this.unitUsedOnDistanceAxis, i.e. depending on the length of the path, either km or m)
+         * @returns The height in meter at the given distance
+         */
+        getHeightAtDist(dist) {
+            // Find lowerIndex, such that dist[lowerIndex] <= dist <= dist[lowerIndex + 1] = dist[higherIndex]
+            // The distance increase between each index is approximately regular, so we can make a first guess
+            // with a complexity of O(1) making sure the dist we are searching is within possible values (>= 0 and <= maxDist)
+            const distToSearch = Math.min(Math.max(dist, 0), this.profileData.maxDist)
+            let lowerIndex = Math.trunc(
+                ((this.profileData.length - 1) * distToSearch) / this.profileData.maxDist
+            )
+            lowerIndex = Math.min(lowerIndex, this.profileData.length - 1)
+            // As it is not perfectly regular (maybe as a result of trunking the decimals in the backend?) we still need to correct our first guess with these two while loops. */
+            while (distToSearch < this.profileData.points[lowerIndex].dist) {
```

Comment:
I would add guard here to avoid negative index due to bad data
```javascript
while (lowerIndex > 0 && distToSearch < this.profileData.points[lowerIndex].dist)
```

Reply:
That's also some relic of `mf-geoadmin3`, I don't mind that we enhance this piece of code in the future, but right now it's really low on our priority list. So if you don't mind, we can write a ticket for post MVP task to have a look into optimizing / fixing this part of the code, but for now we can live with it